packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx

```diff
@@ -3403,8 +3403,7 @@ describe('shallow', () => {
 
     context('updating state', () => {
       // NOTE: There is a bug in react 16 shallow renderer where prevContext is not passed
-      // into componentDidUpdate. Skip this test for react 16 only. add back in if it gets fixed.
```

Comment:
prevContext *should* be passed into componentDidUpdate; is react not going to fix this?

Reply:
Yes. #1139 is a PR for that.
You can see the change at [Unreleased] section in CHANGELOG.md 

https://github.com/facebook/react/blob/master/CHANGELOG.md

Reply:
Then if that's fixed, why do we need an option in the adapter?

Reply:
The PR is to support that componentDidUpdate no longer receives prevContext argument, which is not only ShallowRenderer.
This PR is to support that ShallowRenderer no longer call componentDidUpdate.
These are different fixes so I added the options each other.

Reply:
Why does componentDidUpdate not receive a prevContext argument anymore? It seems like it should.

Reply:
The reason seems to be the following. cc: @bvaughn

> Even with that parameter present- if context isn't managed in the same way as props and state (PR #8655) then it could be inconsistent between the instance and the fiber.

> Update: Chatted in meatspace with @acdlite about this. He pointed out another existing stack limitation with fiber WRT shouldComponentUpdate. (Seems like context is already flaky but fiber will make it more so.) The only safe way to use context is to pass singleton/static props (eg the way react-redux Provider passes down store) that components subscribe to for changes.

https://github.com/facebook/react/pull/8631#issuecomment-270450606

Reply:
Wow, that seems like a pretty major change to the way context works then.

I've not found context to be flaky at all in React <= 15.

@lelandrichardson any suggestions for a better name for this part of the adapter interface?

Reply:
> Why does componentDidUpdate not receive a prevContext argument anymore? It seems like it should.

@ljharb Check out facebook/react/pull/8631 for a description of the options we considered.

> The only safe way to use context is to pass singleton/static props...that components subscribe to for changes.

Yes. That sounds accurate.